### PR TITLE
chore(DTFS2-6997) - replace moment in gef

### PR DESCRIPTION
**Pull request by MarRobSoftwire**
## Introduction :pencil2:
Moment js is deprecated, it should be replaced. This PR will not cover removing moment from tests - see #2761 .

## Resolution :heavy_check_mark:
* Replace moment() with vanilla Dates where possible and date-fns where necessary in gef-ui

## Misc
* update test config and linting for ts files in gef-ui

### CODE REVIEW

File: gef-ui/server/utils/display-items.js [hunk 1]
```diff
@@ -82,21 +82,27 @@ const facilityItems = (facilityUrl, {
...
+      method: (value) => (isTrueSet(value) ? STAGE.ISSUED : STAGE.UNISSUED),
     },
     {
       label: 'Date issued to exporter',
       id: 'issueDate',
-      method: (callback) => moment(callback)
-        .format('D MMMM YYYY'),
+      method: (value) => {
+        // issueDate is an ISO-8601 string with milliseconds (e.g '2024-02-14T00:00:00.000+00:00')
+        const date = new Date(value);
```
Comment (MarRobSoftwire) [inline gef-ui/server/utils/display-items.js:92]: use a date-fns helper to make it clear parsing from ISO

File: gef-ui/templates/includes/application-preview/task-comments.njk [hunk 2]
```diff
@@ -61,7 +61,7 @@
   <div class="govuk-!-margin-top-3" data-cy="latest-comment">
     {% for comment in comments %}
       <strong>Comments from {{ comment.firstname }} {{ comment.surname }}</strong>
-      <p>{{ comment.createdAt | localiseTimestamp('DD MMM YYYY h:mma', timezone) }}</p>
+      <p>{{ comment.createdAt | localiseTimestamp('dd MMM yyyy h:mmaaa', timezone) }}</p>
```
Comment (MarRobSoftwire) [inline gef-ui/templates/includes/application-preview/task-comments.njk:64]: Sanity check please:
|[moment](https://momentjscom.readthedocs.io/en/latest/moment/04-displaying/01-format/) |  [date-fns](https://date-fns.org/v3.3.1/docs/format)| example|
|---------|--------|---|
|`h` | `h`| 1, 2, 3, ... 11, 12 |
|`mm`| `mm`| 00, 01, ..., 58, 59 |
|`a`|`aaa`| am, pm |

File: gef-ui/api-test.jest.config.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
+  preset: 'ts-jest/presets/js-with-babel',
```
Comment (AlexBramhill) [inline gef-ui/api-test.jest.config.js:2]: `js-with-ts`?

File: gef-ui/jest.config.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
+  preset: 'ts-jest/presets/js-with-babel',
```
Comment (AlexBramhill) [inline gef-ui/jest.config.js:2]: `js-with-ts`?

File: gef-ui/templates/includes/application-preview/task-comments.njk (hunk 2 above)
Comment (AlexBramhill) [reply gef-ui/templates/includes/application-preview/task-comments.njk:64]: lgtm

File: gef-ui/.eslintrc.js [hunk 5]
```diff
@@ -2,35 +2,49 @@ const baseParserOptions = {
...
+  ],
+  'import/no-named-as-default': 'off',
+  'import/prefer-default-export': 'off',
+  'import/extensions': 'off',
+  'implicit-arrow-linebreak': 'off',
+  'import/no-extraneous-dependencies': ['error', { devDependencies: ['**/*.test.{js,ts}', '**/*.spec.{js,ts}', '**/webpack.*.{js,ts}', '**/api-tests/**', '**/__mocks__/**'] }],
+  'object-curly-newline': ['error', {
+    consistent: true,
+  }],
+  'no-restricted-syntax': 'off',
```
[the remark this answers is not in the document]
Comment (MarRobSoftwire) [reply gef-ui/.eslintrc.js:27]: Cant see anything - will remove

File: gef-ui/api-test.jest.config.js (hunk 3 above)
Comment (MarRobSoftwire) [reply gef-ui/api-test.jest.config.js:2]: gef-ui uses Babel so I think it's needed here - can give it a try with `js-with-ts` if you  like?

Comment (MarRobSoftwire) [reply gef-ui/api-test.jest.config.js:2]: unit tests fail without it

Comment (AlexBramhill) [reply gef-ui/api-test.jest.config.js:2]: nope -- lets keep

**Review by AlexBramhill — APPROVED** [review]

**Review by AlexBramhill — APPROVED** [review]

**Pull request merged**